pytorch_lightning/plugins/precision/native_amp.py

```diff
@@ -96,7 +96,8 @@ def pre_optimizer_step(
                 " To request, please file a Github issue in PyTorch and tag @mcarilli"
             )
         result = lambda_closure()  # native amp does not support closures
-        self.scaler.unscale_(optimizer)
+        if not model.automatic_optimization:
```

Comment:
Do we need check isinstance(model, pl.LightningModule) here? Is it possible that model is wrapped ?

Reply:
No, not needed. 1) the methods here are type annotated with LightningModule. 2) we are explicitly passing in `self.lighting_module` where we call the method, and `self.lightning_module` is guaranteed to be the unwrapped pure LM.